### PR TITLE
fix: Properly handle store flags (when replacing)

### PR DESCRIPTION
**Pull request by jameshoulahan**

### CODE REVIEW

File: internal/state/actions.go [hunk 1]
```diff
@@ -436,48 +436,31 @@ func (state *State) actionSetMessageFlags(ctx context.Context, tx *ent.Tx, messa
 	})
 
 	// If setting messages as seen, only set those messages that aren't currently seen.
-	var messagesToApply []imap.MessageID
+	setSeen := make(map[bool][]imap.MessageID)
```
Comment (LBeernaertProton) [inline internal/state/actions.go:439]: `make(map[..][..], 2)`

File: internal/state/actions.go [hunk 2]
```diff
@@ -436,48 +436,31 @@ func (state *State) actionSetMessageFlags(ctx context.Context, tx *ent.Tx, messa
...
+	for seen, messageIDs := range setSeen {
+		if err := state.user.GetRemote().SetMessagesSeen(ctx, messageIDs, seen); err != nil {
 			return err
 		}
 	}
 
-	// If setting messages as flagged, only set those messages that aren't currently flagged.
-	if setFlags.ContainsUnchecked(imap.FlagFlaggedLowerCase) {
-		var messagesToApply []imap.MessageID
+	setFlagged := make(map[bool][]imap.MessageID)
```
Comment (LBeernaertProton) [inline internal/state/actions.go:453]: make(map[..][..], 2)

**Review by LBeernaertProton — CHANGES_REQUESTED** [review]

File: internal/state/actions.go (hunk 1 above)
Comment (jameshoulahan) [reply internal/state/actions.go]: even better, replaced with 
```go
	setSeen := map[bool][]imap.MessageID{true: {}, false: {}}
```

File: internal/state/actions.go (hunk 2 above)
Comment (jameshoulahan) [reply internal/state/actions.go]: replaced with map literal

**Review by LBeernaertProton — APPROVED** [review]

**Pull request merged**